app/views/RoomView/index.js

```diff
@@ -919,14 +921,31 @@ class RoomView extends React.Component {
 			joined, room, selectedMessage, editing, replying, replyWithMention, readOnly
 		} = this.state;
 		const { navigation, theme } = this.props;
+		const { joinCodeRequired } = room;
 
 		if (!this.rid) {
 			return null;
 		}
 		if (!joined && !this.tmid) {
 			return (
 				<View style={styles.joinRoomContainer} key='room-view-join' testID='room-view-join'>
-					<Text accessibilityLabel={I18n.t('You_are_in_preview_mode')} style={[styles.previewMode, { color: themes[theme].titleText }]}>{I18n.t('You_are_in_preview_mode')}</Text>
+					{
+						joinCodeRequired
+							? (
+								<>
+									<Text accessibilityLabel={I18n.t('Insert_Join_Code')} style={[styles.previewMode, { color: themes[theme].titleText }]}>{I18n.t('Insert_Join_Code')}</Text>
+									<TextInput
+										returnKeyType='default'
+										placeholder={I18n.t('Join_Code')}
+										onChangeText={(value) => { this.setState({ joinCode: value }); }}
+										underlineColorAndroid='transparent'
+										containerStyle={styles.joinCodeInput}
+										theme={theme}
+									/>
+								</>
```

Comment:
![Screen Shot 2020-10-07 at 12 55 05 PM](https://user-images.githubusercontent.com/29778115/95355806-593ca880-089c-11eb-9ff5-f5e7b3883729.png)
You can use a Input that cover `screen.width - 32` (margin horizontal = 16), it should be better

Reply:
taking the size of the screen did not work, the size of 200 in width